File: projectEuler/22/22.cpp
#include <fstream>
#include <iostream>
#include <vector>
#include <algorithm>
#include <sstream>

// Function to calculate the name score
int nameScore(const std::string& name) {
    int score = 0;
    for (char c : name) {
        score += c - 'A' + 1;
    }
    return score;
}

int main() {
    std::string path = "your txt file path";
    std::ifstream file(path);
    std::string data;
    std::getline(file, data);
    file.close();

    // Replace quotes and split by comma
    std::vector<std::string> names;
    std::stringstream ss(data);
    std::string name;
    while (std::getline(ss, name, ',')) {
        names.push_back(name.substr(1, name.length() - 2)); // Remove quotes
    }

    // Sort names alphabetically
    std::sort(names.begin(), names.end());

    // Calculate total of all name scores
    long long totalScore = 0;
    for (size_t i = 0; i < names.size(); ++i) {
        totalScore += (i + 1) * nameScore(names[i]);
    }

    std::cout << totalScore << std::endl;

    return 0;
}

File: projectEuler/22/22.py
#Using names.txt (right click and 'Save Link/Target As...'), a 46K text file containing over five-thousand first names, begin by sorting it into alphabetical order. Then working out the alphabetical value for each name, multiply this value by its alphabetical position in the list to obtain a name score.
#For example, when the list is sorted into alphabetical order, COLIN, which is worth 3 + 15 + 12 + 9 + 14 = 53, is the 938th name in the list. So, COLIN would obtain a score of 938 * 53 = 49714.

file_txt="your txt path"
def name_score(name):
    return sum([ord(c) - ord('A') + 1 for c in name])

def main():
    with open(file_txt) as f:
        names = f.read().replace('"', '').split(',')
    names.sort()
    print(sum([(i + 1) * name_score(name) for i, name in enumerate(names)]))

if __name__ == '__main__':
    main()
